internal_worker.go

```diff
@@ -160,6 +161,31 @@ func ensureRequiredParams(params *workerExecutionParameters) {
 	}
 }
 
+// verifyDomainExist does a DescribeDomain operation on the specified domain
+// It returns an error, if the server returns an EntityNotExist or BadRequest error
+// On any other transient error, this method will just return success
+func verifyDomainExist(client m.TChanWorkflowService, domain string, logger *zap.Logger) error {
+
+	ctx, cancel := newTChannelContext()
+	defer cancel()
+
+	_, err := client.DescribeDomain(ctx, &shared.DescribeDomainRequest{Name: &domain})
+	if err != nil {
+		if _, ok := err.(*shared.EntityNotExistsError); ok {
+			logger.Error("domain does not exist", zap.String("domain", domain), zap.Error(err))
+			return err
+		}
+		if _, ok := err.(*shared.BadRequestError); ok {
+			logger.Error("domain does not exist", zap.String("domain", domain), zap.Error(err))
+			return err
+		}
+		// on any other error, just return true
```

Comment:
the assumption is - all other errors are transient. That's the server side contract. The server, for the most part, also wrap all these other errors within InternalServiceError{}. In the case of a transient error, there is not much we can do. Following the idl contract:
```
  shared.DescribeDomainResponse DescribeDomain(1: shared.DescribeDomainRequest describeRequest)
    throws (
      1: shared.BadRequestError badRequestError,
      2: shared.InternalServiceError internalServiceError,
      3: shared.EntityNotExistsError entityNotExistError,
    )
```